src/Dfc.CourseDirectory.FindACourseApi/Controllers/CoursesController.cs

```diff
@@ -22,59 +22,43 @@ public class CoursesController : ControllerBase, IActionFilter
             _log = logger;
         }
 
-        [Route("~/coursesearch")]
-        [HttpPost]
-        [ProducesResponseType(typeof(Features.CourseSearch.ViewModel), StatusCodes.Status200OK)]
+        [HttpPost("~/coursesearch")]
+        [ProducesResponseType(typeof(Features.CourseSearch.CourseSearchViewModel), StatusCodes.Status200OK)]
         [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
         [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-        public async Task<ActionResult> CourseSearch([FromBody] Features.CourseSearch.Query request)
+        public async Task<IActionResult> CourseSearch([FromBody] Features.CourseSearch.Query request)
         {
-            var response = await _mediator.Send(request);
-            return new OkObjectResult(response);
+            var result = await _mediator.Send(request);
+
+            return result.Match<IActionResult>(
+                p => {
+                    _log.LogWarning($"{nameof(CourseSearch)} failed. {nameof(p.Title)}: {{{nameof(p.Title)}}}, {nameof(p.Detail)}: {{{nameof(p.Detail)}}}.", p.Title, p.Detail);
```

Comment:
Do we care about logging this?

Reply:
From an operational perspective, I think this will give us some interesting insights into any consumers who are misusing the API.